core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -248,6 +268,15 @@ abstract class Chunk[+O] extends Serializable {
       buf.result
     }
 
+  /**
+    * Returns true if this chunk is known to have elements of type `B`.
+    * This is determined by checking if the chunk type mixes in `Chunk.KnownElementType`.
+    */
+  def knownElementType[B](implicit classTag: ClassTag[B]): Boolean = this match {
+    case ket: Chunk.KnownElementType[_] if ket.elementClassTag eq classTag => true
+    case _                                                                 => false
+  }
```

Comment:
I'm really not sure this is actually enough, since wouldn't `ByteVectorChunk` have the same element type as `ByteChunk` and `ByteBufferChunk`?